### PR TITLE
Add OpenTelemetry tracer

### PR DESCRIPTION
**Pull request by peterdeme**
For the trace dropping, I've tried multiple things:

- custom sampler: didn't work. we dropped the parent span, but couldn't access child spans.
- don't call `span.End()` for dropped spans - this works but children spans still show up in the UI
- filter spans in `otel-collector.yml` - same issue, only drops the parent span

### CODE REVIEW

File: tracing/opentelemetry/opentelemetry.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	// The main issue is that even if we drop a parent span, the child spans will still be recorded.
+
+	for key, value := range internal.DeduplicateFields(fields) {
+		span.SetAttributes(attribute.String(key, fmt.Sprintf("%v", value)))
+	}
+
+	if err != nil {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, "")
+	}
```
Comment (ilya-hontarau) [inline tracing/opentelemetry/opentelemetry.go:55]: isnt it a deadcode?

Comment (peterdeme) [reply tracing/opentelemetry/opentelemetry.go:55]: nope

**Comment by ilya-hontarau** [comment]
> For the trace dropping, I've tried multiple things:

I think the only thing we can do is to create an issue in the otel sdk?
https://github.com/open-telemetry/opentelemetry-go/issues

**Review by ilya-hontarau — APPROVED** [review]

**Comment by peterdeme** [comment]
@ilya-hontarau I'll probably do but keep in mind that this is a fairly serious standard, so all SDKs would need to follow and implement it, so I don't have high hopes

**Pull request merged**